protocol/messenger.go

```diff
@@ -6259,16 +6259,18 @@ func (m *Messenger) GetMentionsManager() *MentionManager {
 	return m.mentionsManager
 }
 
-func (m *Messenger) getMessagesToDelete(message *common.Message, chatID string) ([]*common.Message, error) {
+func (m *Messenger) getConnectedMessages(message *common.Message, chatID string) ([]*common.Message, error) {
 	var messagesToDelete []*common.Message
 	// In case of Image messages, we need to delete all the images in the album
 	if message.ContentType == protobuf.ChatMessage_IMAGE {
 		image := message.GetImage()
-		messagesInTheAlbum, err := m.persistence.albumMessages(chatID, image.GetAlbumId())
-		if err != nil {
-			return nil, err
+		if image != nil && image.AlbumId != "" {
+			messagesInTheAlbum, err := m.persistence.albumMessages(chatID, image.GetAlbumId())
+			if err != nil {
+				return nil, err
+			}
+			messagesToDelete = append(messagesToDelete, messagesInTheAlbum...)
 		}
```

Comment:
I think we need an `else` here to append the original `message`. So if we have no album, we just send the simple message so that it can be edited or deleted.

If you want to reduce duplication, you can do something like this instead:

```go
func (m *Messenger) getConnectedMessages(message *common.Message, chatID string) ([]*common.Message, error) {
	var messagesToDelete []*common.Message
	// In case of Image messages, we need to delete all the images in the album
	if message.ContentType == protobuf.ChatMessage_IMAGE {
		image := message.GetImage()
                if image != nil && image.AlbumId != "" {
			messagesInTheAlbum, err := m.persistence.albumMessages(chatID, image.GetAlbumId())
			if err != nil {
				return nil, err
			}
			messagesToDelete = append(messagesToDelete, messagesInTheAlbum...)
                        return messagesToDelete
		}
         }
         return append(messagesToDelete, message), nil
}
```

Also, rename `messagesToDelete` to `connectedMessages` :sweat_smile: